### PR TITLE
Remove `canOptimizeEdit` from Z code generator

### PR DESCRIPTION
**Pull request by pparkkin**
This function was left over from early compiler technology. It is not
used and should be removed.

Issue https://github.com/eclipse/omr/issues/1397

Signed-off-by: Paavo Parkkinen <pparkkin@gmail.com>

### CODE REVIEW

**Comment by pparkkin** [comment]
Should this be tested in System z?

**Comment by 0xdaryl** [comment]
IBM will take care of System Z testing because at the moment such systems are not part of CI.  Thanks for your contribution!

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -11416,14 +11416,6 @@ bool TR_S390Peephole::forwardBranchTarget()
       }
    return false;
    }
-bool
-OMR::Z::CodeGenerator::canOptimizeEdit(uint8_t *parm, int32_t length)
-   {
-   bool leadingSigs = false, trailingStr = false;
-   uint8_t fill = 0;
-   int32_t numDigits = 0, numSigs = 0, sigOff = 0, lastDigit = 0, digitsAfterSig = 0;
-   return self()->parseEditParm(parm, length, leadingSigs, trailingStr, fill, numDigits, numSigs, sigOff, lastDigit, digitsAfterSig);
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: `parseEditParm` is actually dead too.  I should have mentioned that in the issue.  Would you mind removing that as part of this PR as well?

Comment (pparkkin) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: Removed.

Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: Excellent!  Thanks.

**Pull request merged**